### PR TITLE
 glibc: always compile with a gcc stdenv

### PR DESCRIPTION
**Pull request by sternenseemann**
Our glibc doesn't compile with clang and useLLVM doesn't make the stdenv
implicitly switch to musl (and arguably shouldn't do that). To make LLVM
cross compilation without patching glibc to compile with clang (which is
probably a lot of work), always compile glibc with gcc. This should
unbreak a lot of cross compilation tests, I have tested:

* tests.cross.llvm.gnu64
* tests.cross.llvm.aarch64-multiplatform

Basing against staging because this change only really makes sense if
useLLVM works with llvmPackages_7 which is only the case in staging at
the moment.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ericson2314 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10257,10 +10257,12 @@ in
...
 
-  crossLibcStdenv = overrideCC stdenv
-    (if stdenv.hostPlatform.useLLVM or false
-     then buildPackages.llvmPackages_8.lldClangNoLibc
-     else buildPackages.gccCrossStageStatic);
+  gccCrossLibcStdenv = overrideCC stdenv buildPackages.gccCrossStageStatic;
+
+  crossLibcStdenv =
+    if stdenv.hostPlatform.useLLVM or false
+    then overrideCC stdenv buildPackages.llvmPackages_8.lldClangNoLibc
```
Comment (Ericson2314) [inline pkgs/top-level/all-packages.nix:10264]: BTW this should use the default LLVM version now that it's updated to be newer for cross.

Comment (sternenseemann) [reply pkgs/top-level/all-packages.nix:10264]: It's still clang 7 for `useLLVM` right? I think we'll have to check up on this topic anyways. Unfortunately there's still *some* issue with `llvmPackages_7.lldClang` with C++ and libunwind (i. e. `ncurses`' C++ library doesn't compile with `useLLVM` atm).

This is probably something unrelated to this though, I'd expect we can compile musl with clang 7 just fine. I'll try to rememeber to check that later.

Comment (Ericson2314) [reply pkgs/top-level/all-packages.nix:10264]: I think it's a later version when cross compilation is involved. We can tweak the condition if something isn't working.

Comment (sternenseemann) [reply pkgs/top-level/all-packages.nix:10264]: At least `stdenv` itself still uses clang 7.

**Pull request merged**